File: newsroom/lib/publish.ts
/**
 * Publish the finished edition to the live site: POST it to the worker, which
 * writes KV (edisi:current). The site reads that at runtime — no rebuild. This
 * is loop 3's "update a real system" step. Needs AKSARA_URL + EDISI_TOKEN.
 */
export async function publishEdisi(edisi: unknown): Promise<boolean> {
  const url = (process.env.AKSARA_URL ?? process.env.PUBLIC_AKSARA_URL)?.replace(/\/$/, '');
  const token = process.env.EDISI_TOKEN;
  if (!url || !token) {
    console.log('[publish] AKSARA_URL / EDISI_TOKEN belum diset — edisi tidak dikirim (kering).');
    return false;
  }
  try {
    const r = await fetch(`${url}/edisi`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json', Authorization: `Bearer ${token}` },
      body: JSON.stringify(edisi),
      signal: AbortSignal.timeout(15_000),
    });
    console.log(`[publish] /edisi -> ${r.status}`);
    return r.ok;
  } catch (e) {
    console.error('[publish] gagal:', e);
    return false;
  }
}

File: newsroom/publish.py
"""
Publish the finished edition to the live site: POST it to the worker, which writes
KV (edisi:current). The site reads that at runtime, no rebuild (loop 3's "update a
real system"). Needs AKSARA_URL + EDISI_TOKEN; without them it runs dry (logs only)
so local and CI dry-runs never touch production. Ported from newsroom/lib/publish.ts.
"""

from __future__ import annotations

import os

import httpx

from .models import Edisi


def publish_edisi(edisi: Edisi) -> bool:
    url = (os.environ.get("AKSARA_URL") or os.environ.get("PUBLIC_AKSARA_URL") or "").rstrip("/")
    token = os.environ.get("EDISI_TOKEN")
    payload = edisi.model_dump(exclude_none=True)

    if not url or not token:
        print("[publish] AKSARA_URL / EDISI_TOKEN belum diset, edisi tidak dikirim (kering).")
        return False
    try:
        r = httpx.post(
            f"{url}/edisi",
            json=payload,
            headers={"Authorization": f"Bearer {token}"},
            timeout=httpx.Timeout(15.0),
        )
        print(f"[publish] /edisi -> {r.status_code}")
        return r.is_success
    except Exception as e:  # noqa: BLE001
        print(f"[publish] gagal: {e}")
        return False
